From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Tue, 1 Sep 2026 16:25:45 +0200
Subject: fix(busybox): do not install nbd-client applet

The busybox `nbd-client` applet does not support the option `-check` and
others. This causes the test 60-NFS to fail with busybox.

So not install the `nbd-client` applet.

Fixes: https://github.com/dracut-ng/dracut/issues/2677
Origin: upstream, https://github.com/dracut-ng/dracut/commit/c051f59e8d8b14c8e7454323ddb742dda382bef1
---
 modules.d/10busybox/module-setup.sh | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/modules.d/10busybox/module-setup.sh b/modules.d/10busybox/module-setup.sh
index 9df42c8..f175f72 100755
--- a/modules.d/10busybox/module-setup.sh
+++ b/modules.d/10busybox/module-setup.sh
@@ -38,6 +38,11 @@ install() {
             # See https://github.com/vda-linux/busybox_mirror/issues/34
             continue
         fi
+        if [[ ${_path##*/} == nbd-client ]]; then
+            # The busybox nbd-client applet does not support the option -check.
+            # See https://github.com/vda-linux/busybox_mirror/issues/35
+            continue
+        fi
 
         # if busybox is built without CONFIG_FEATURE_INSTALLER=y, the list
         # has only plain names
